src/store/modules/search.store.js

```diff
@@ -174,18 +181,37 @@ const actions = {
 
                 // Automatically select the feature
                 try {
-                    getFeature(entry.layer, entry.featureId, rootState.position.projection, {
-                        lang: rootState.i18n.lang,
-                        screenWidth: rootState.ui.width,
-                        screenHeight: rootState.ui.height,
-                        mapExtent: flattenExtent(getters.extent),
-                        coordinate: entry.coordinate,
-                    }).then((feature) => {
+                    if (entry.layer.getTopicForIdentifyAndTooltipRequests) {
+                        getFeature(entry.layer, entry.featureId, rootState.position.projection, {
+                            lang: rootState.i18n.lang,
+                            screenWidth: rootState.ui.width,
+                            screenHeight: rootState.ui.height,
+                            mapExtent: flattenExtent(getters.extent),
+                            coordinate: entry.coordinate,
+                        }).then((feature) => {
+                            dispatch('setSelectedFeatures', {
+                                features: [feature],
+                                dispatcher,
+                            })
+                        })
+                    } else {
+                        // For imported KML and GPX files
+                        let features = []
+                        if (entry.layer.type === LayerTypes.KML) {
+                            features = parseKml(entry.layer, rootState.position.projection, [])
+                        }
+                        if (entry.layer.type === LayerTypes.GPX) {
+                            features = parseGpx(entry.layer, rootState.position.projection, [])
+                        }
+                        // TODO see if there is a better way of finding the feature
```

Comment:
as you are in the store module, and we try to avoid at all cost to link the map to the store : no I don't think so.
Or you would have to do that somewhere else where you have access to the map and the store (meaning in a component), but I bet it makes more sense to have that alongside all the other search types.

Reply:
Ok thanks then i will leave it like this